src/org/omegat/util/gui/FontFallbackManager.java

```diff
@@ -52,15 +55,97 @@ private FontFallbackManager() {
      * <li><a href="https://github.com/googlei18n/noto-emoji">Noto Emoji</a>
      * </ul>
      */
-    private static final Set<String> FONT_BLACKLIST = Collections.singleton("Apple Color Emoji");
-    private static final Font FONT_UNAVAILABLE = new Font("", 0, 0);
+    private static final Set<String> FONT_BLACKLIST = new HashSet<>(
+            Arrays.asList("Apple Color Emoji", "Noto Color Emoji")
+    );
+    private static final Font FONT_UNAVAILABLE = new Font("", Font.PLAIN, 0);
 
     private static final Logger LOGGER = Logger.getLogger(FontFallbackManager.class.getName());
 
     private static final Font[] RECENT_FONTS = new Font[8];
     private static int lastFontIndex = 0;
     private static final Map<Integer, Font> CACHE = new ConcurrentHashMap<>();
 
+    private static final FontRenderContext DEFAULT_CONTEXT = new FontRenderContext(null, false, false);
+
+    private FontFallbackManager() {
+    }
+
+    /**
+     * Detect specified codePoints can display with specified font.
+     */
+    public static int canDisplayUpTo(Font font, String str) {
```

Comment:
It looks like you copied these implementations from the `Font` class. That's fine if that's really the only way to do it, but it would be good to mention that fact in the comments.

Reply:
Could you propose another way to improve here? If FontFallbackListener don't use canDisplayUpTo method, it can be simple. I don't have idea around here just now.

Reply:
Yeah I don't have a good idea to improve it.

My concern with duplicating standard library code is that we could miss improvements or changes across Java versions.

Maybe it would be good to only use our version when the OS is Mac:

```java
if (!Platform.isMacOSX()) {
    return font.canDisplayUpTo(...);
} else {
    // This implementation is copied from java.awt.Font; we use it to work
    // around a bug on macOS whereby all fonts always claim to support all
    // codepoints; see bug#1051 (https://sourceforge.net/p/omegat/bugs/1051/)
    ...
}

```

Reply:
I realized that we can be smarter about when we use the workaround code; see https://github.com/omegat-org/omegat/pull/118/commits/c7e3b3bdbd9c92b4cfd8d490cfc38d96522c25c4